src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -25,7 +25,27 @@ namespace OpenTelemetry.AutoInstrumentation.Logger;
 
 internal static class LogBuilderExtensions
 {
+    private static bool? _autoInstrumentationStartupAssemblyConfigured;
     private static Type? _loggingProviderSdkType;
+    private static Type? _applicationLifetimeType;
+
+    public static void AddOpenTelemetryLogsFromIntegration(this ILoggingBuilder builder)
+    {
+        // For Net6, if HostingStartupAssembly is configured, we don't want to call integration again for host's ServiceCollection.
+        // OpenTelemetryLogger-related services were already added to WebApplicationServiceCollection and will
+        // be copied to host's ServiceCollection later. We can't depend on integration's
+        // capability to detect if integration was called before (by checking if ServiceDescriptor with
+        // given type is already added to ServiceCollection) as copying services from WebApplicationServiceCollection
+        // to host's ServiceCollection happens AFTER integration is called.
+
+        // All of this additional checking is NOT needed for net7. There we can rely on integration's capability
+        // to detect if integration was called before, because WebApplicationServiceCollection is not used when building host.
+
+        if (builder.Services is ServiceCollection && !(IsNet6() && IsAutoInstrumentationStartupAssemblyConfigured() && IsHostServiceCollection(builder.Services)))
```

Comment:
Should we only check `IsNet6` and `IsAutoInstrumentationStartupAssemblyConfigured`, and use a static bool to ensure the integration has already been done? If static doesn't work, we could also rely on a temporary environment variable

Reply:
I understand, but not comfortable in using this check `builder.Services is ServiceCollection` as we know this has caused issues earlier and we released a minor version for it.